src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,130 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+@Embeddable
+public class Sections {
+	private final int NOT_FOUND = -1;
+
+	@OneToMany(mappedBy = "line", cascade = CascadeType.ALL, orphanRemoval = true)
+	private List<Section> sections = new ArrayList<>();
+
+	public Sections() {
+	}
+
+	public List<Station> getStations() {
+		List<Station> downStations = sections.stream()
+			.map(Section::getDownStation)
+			.collect(Collectors.toList());
+
+		Section firstSection = sections.stream()
+			.filter(it -> !downStations.contains(it.getUpStation()))
+			.findFirst()
+			.orElseThrow(IllegalArgumentException::new);
+
+		List<Station> stations = new ArrayList<>();
+		stations.add(firstSection.getUpStation());
+		stations.add(firstSection.getDownStation());
+
+		Optional<Section> nextSection = findNextSection(firstSection.getDownStation());
+
+		while(nextSection.isPresent()) {
+			Station station = nextSection.get().getDownStation();
+
+			stations.add(station);
+			nextSection = findNextSection(station);
+		}
+
+		return stations;
+	}
+
+	public void add(Line line, Station upStation, Station downStation, int distance) {
+		if(sections.isEmpty()) {
+			sections.add(new Section(line, upStation, downStation, distance));
+			return;
+		}
+
+		validateDuplication(upStation, downStation);
+		validateInclusion(upStation, downStation);
+
+		updateSectionInMiddle(line, upStation, downStation, distance);
+		sections.add(new Section(line, upStation, downStation, distance));
+	}
+
+	private void validateDuplication(Station upStation, Station downStation) {
+		sections.stream()
+			.filter(it -> it.isDuplicated(upStation, downStation))
+			.findFirst()
+			.ifPresent(it -> { throw new IllegalArgumentException(); });
+	}
+
+	private void validateInclusion(Station upStation, Station downStation) {
+		sections.stream()
+			.filter(it -> it.isInclude(upStation) || it.isInclude(downStation))
+			.findFirst()
+			.orElseThrow(IllegalArgumentException::new);
+	}
+
+	private void updateSectionInMiddle(Line line, Station upStation, Station downStation, int distance) {
+		int index = searchUpStationOrDownStation(upStation, downStation);
+
+		if(index == NOT_FOUND) {
+			return;
+		}
+
+		Section foundSection = sections.get(index);
+		validateDistance(foundSection, distance);
+
+		/** A - B
+		 *  A  ---  C
+		 */
+		if(foundSection.isUpStation(upStation)) {
```

Comment:
전체적으로 `if`, `while`등에 대한 컨벤션 한번 확인 해주시면 감사하겠습니다 😄 
> https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

```suggestion
		if (foundSection.isUpStation(upStation)) {
```

Reply:
아!?!? 무슨 일이 발생한걸까요... 확인해보겠습니다 😅😅